javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -457,6 +475,16 @@ public boolean containsKey(K key) {
     }
 
     @Override
+    public Option<Tuple2<K, V>> max() {
+        return entries.max();
+    }
+
+    @Override
+    public Option<Tuple2<K, V>> min() {
+        return entries.min();
```

Comment:
is this the same as `head`, as `last` was `max`?
If yes, please express it in code also (i.e. reuse).